### ISSUE TITLE
Blazor EditForm does not bind to Model when using FluentButton

### ISSUE THREAD

**Issue by htmlsplash** [microsoft/fluentui-blazor#3015]
# 🐛 Bug Report

Blazor EditForm does not populate my model's data (when it is submitted) present in the  Fluent Text Fields when submitting the form using Fluent Button (of type submit).

## 💻 Repro or Code Sample

See attached file for all the code.

## 🤔 Expected Behavior

When you submit the form, the Model should be populated from user's input provided by the fluent text input fields.

## 😯 Current Behavior

The model doesn't not have its fields populated from user input when I submit the form.

## 💁 Possible Solution

n/a

## 🔦 Context

Basic form editing/submission is not working.

## 🌍 Your Environment

.net 9
Blazor Web app: Default is Static SSR, but test page uses Interactive Server render mode (via render mode directive).
Running in VS (latest) on Windows 11.
Browser: Chrome
Latest version of Fluent UI Blazor library via nuget.

[fluent_ex.txt](https://github.com/user-attachments/files/18015073/fluent_ex.txt)

**Comment by vnbaaij** [comment]
You have not followed our issue template and have not supplied enough detail for us to investigate. We will not work on this until details are supplied. If not done in 3 days, we will close the issue. 


What version of the library? What kind of project (WASM/Web App)? What .NET version? What rendermode does the page use?

If you want us to help **you**, you need to help **us**!

**Comment by htmlsplash** [comment]
I have updated the posting. Let me know if you need more information.

**Comment by vnbaaij** [comment]
I created a new Fluent Blazor Web App project with 'Interactive rendermode' set to 'None' and 'Include sample pages' checked. Replaced `Counter.razor` with your attached sample code. Top of the page looks like this (as this was not in your attachment) :
```
@page "/counter"
@using System.ComponentModel.DataAnnotations
@rendermode InteractiveServer

<PageTitle>Counter</PageTitle>
```

When running this and filling out the form, I get:
![Image](https://github.com/user-attachments/assets/2c14346f-655c-4379-8fbf-98e15f8a2144)

As you can see in the console, the model data is there. 

Are you sure your page is using Server rendermode? If it is SSR, check the notes at https://www.fluentui-blazor.net/Forms

Maybe you can share all your code in a repo so we can reproduce the issue?

**Comment by htmlsplash** [comment]
Guys, I found the problem, and it is on me. I was using static SSR on this page. What happened was, I was copying/pasting from another form and forgot to copy the render mode attribute. By adding the render mode attribute "Interactive Server" it works. Interestingly, the hot reload in VS is bugged in my project too which doesn't help. Saving the page with changes did not work, I had to recompile and refresh the page.

Because without the interactive server attribute, I was using static SSR for the page. By following the instructions on the sample page, I added the Name attribute with the required values to my form input fields and that work. 
Q: Is there a reason why you guys don't generate the Name attribute on the input field on the client side?

FYI:
You have spelling mistake in the docs, in the Note section:

It reads: "Note: When using the input components in SSR mode, be sure to **assing** a value to the Name property"
The word "assing" should read  "assign".

Sorry about the hassle, thanks for your help.
